src/shogun-core-main/src/main/java/de/terrestris/shoguncore/service/TreeNodeService.java

```diff
@@ -66,7 +65,7 @@ public E cloneAndPersistTreeNode(E node) throws Exception {
                 // recursive call for all children
                 clonedChildren.add(cloneAndPersistTreeNode(childNode));
             }
-            ((TreeFolder) node).setChildren((List<TreeNode>) children);
+            ((TreeFolder) node).setChildren((List<TreeNode>) clonedChildren);
```

Comment:
I think @annarieger and I may recently have ran into exactly the problem you fixed here, :top:

Reply:
:pray: I hope it'll fix the bug while cloning of applications